perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -890,485 +792,37 @@ def _CliBasedTests(output_results, metadata, vm, iteration_count,
-    vm.RemoteCommand('sudo pip install python-swiftclient')
-    vm.bucket_name = 'pkb%s' % FLAGS.run_uri
-    self.swift_command_prefix = _MakeSwiftCommandPrefix(vm.client.url,
-                                                        vm.client.tenant,
-                                                        vm.client.user,
-                                                        vm.client.password)
-    vm.RemoteCommand('swift %s post %s' % (self.swift_command_prefix,
-                                           vm.bucket_name))
-
-  def Run(self, vm, metadata):
-    """Run upload/download on vm with swift cli.
-
-    Args:
-      vm: The vm being used to run the benchmark.
-      metadata: the metadata to be stored with the results.
-
-    Returns:
-      A list of lists containing results of the tests. Each scenario outputs
-      results to a list of the following format:
-        name of the scenario ,value, unit of the value, metadata
-        e.g.,
-        'one byte object upload latency p50', 0.800, 'seconds', 'storage=swift'
-
-      Then the final return value is the list of the above list that reflect
-      the results of all scenarios ran here.
-    """
-    metadata[SWIFTCLIENT_LIB_VERSION] = _GetClientLibVersion(
-        vm, 'python-swiftclient')
-    results = []
-    scratch_dir = vm.GetScratchDir()
-
-    # CLI tool based tests
-
-    clean_up_container_cmd = ('swift %s delete %s'
-                              % (self.swift_command_prefix, vm.bucket_name))
-    upload_cmd = ('time '
-                  'swift %(prefix)s upload %(container)s %(directory)s'
-                  % {'container': vm.bucket_name,
-                     'directory': '%s/run/data/' % scratch_dir,
-                     'prefix': self.swift_command_prefix})
-
-    clean_up_local_temp_cmd = 'rm -r %s/run/temp/*' % scratch_dir
-    download_cmd = ('time '
-                    'swift %(prefix)s download %(container)s -D %(directory)s'
-                    % {'container': vm.bucket_name,
-                       'directory': '%s/run/temp/' % scratch_dir,
-                       'prefix': self.swift_command_prefix})
-
-    iteration_count = (CLI_TEST_ITERATION_COUNT
-                       if FLAGS.cli_test_size == 'normal'
-                       else LARGE_CLI_TEST_ITERATION_COUNT)
-
-    _CliBasedTests(results, metadata, vm, iteration_count,
-                   clean_up_container_cmd, upload_cmd, clean_up_local_temp_cmd,
-                   download_cmd)
-
-    if FLAGS.object_storage_scenario in ('all', 'api_data',):
-      logging.warn('Skipping API-based scenario for object storage service. '
-                   'Reason: Not yet implemented for OpenStack Swift')
-
-    return results
+  # Prepare data on vm, create a run directory on scratch drive, and add
+  # permission.
+  scratch_dir = vm.GetScratchDir()
+  vm.RemoteCommand('sudo mkdir -p %s/run/' % scratch_dir)
+  vm.RemoteCommand('sudo chmod 777 %s/run/' % scratch_dir)
```

Comment:
The changes failed in my testing, and I realized why we need `777`. Because we're using `sudo`, the directories will be created with owner `root` and group whatever group `root` is in. But later on, we want to run commands as PKB's user. Since we can't guarantee that PKB's user is in the same group as `root`, we have to allow `rwx` access for non-group members, which winds up being permissions `777`.

I think ideally we would let the PKB user own the directories, but that is more complexity. I'd like to leave this as `777` for now.